kik_unofficial/client.py

```diff
@@ -890,6 +894,26 @@ def data_received(self, data: bytes):
         if not self.cleanup_task:
             self.cleanup_task = self.loop.call_later(self.cleanup_interval, self.cleanup_buffer)
 
+    @staticmethod
+    def split_elements(data_bytes):
```

Comment:
Again, this is something that `BeautifulSoup` already does, so if you want to be more elegant we can handle multiple XML elements inside `on_new_data_received` or in a different function

Reply:
Also, what about future elements that are not `k`, `pong`, `message`, or `stc`?

Reply:
I have gone over this again after using it for a few weeks on my bot. There were some errors, but only because it was not catching a second <message> element attached to the second part of multi-packet data. 

I made some changes so that the split happens after we match messages for multi-packet messages. If they are not multi packet we then split up in to multiple message elements and send to the loop.

BeautifulSoup can do something similar yes. I am unsure how to implement it to split the xml elements in to the correct groups so i am using regex. Currently the elements it is looking for are the only elements that get sent from the KIK servers. If there are more in the future we can add different regex.

Reply:
Yes this is better